lib/rex/ui/text/shell.rb

```diff
@@ -350,6 +350,40 @@ def print(msg='')
 
 protected
 
+  #
+  # Get a single line of input, following continuation directives as necessary.
+  #
+  def get_input_line
+    line = "\\\n"
+    prompt_needs_reset = false
+
+    self.cont_flag = false
+    while line =~ /(^|[^\\])\\\s*$/
+      # Strip \ and all the trailing whitespace
+      line.sub!(/\\\s*/, '')
+
+      if line.length > 0
+        # Using update_prompt will overwrite the primary prompt
+        input.prompt = output.update_prompt(self.cont_prompt)
+        self.cont_flag = true
+        prompt_needs_reset = true
+      end
+
+      output.input = input
+      line << input.pgets()
```

Comment:
does this call to pgets require the () for some reason or is this your python leaking through?  ;)

Reply:
Hah, my Python would never leak through! That was left over from the original `line = input.pgets()` statement. I'd be happy to update it if it's an eye sore.

Reply:
@kernelsmith: thx for diving in.
@ZeroSteiner: My own $0.02 is to make it look consistent. I find these python leaks every once in a while and do the cleanup inline with some other commit (I don't have h00die's chutzpah these days for refactor PRs, props to them). They're not a problem IMO, but novice coders might think its some obscure ruby magic, or adopt the habit themselves...